src/qt/diagnosticsdialog.cpp

```diff
@@ -0,0 +1,375 @@
+#include "diagnosticsdialog.h"
+#include "ui_diagnosticsdialog.h"
+
+DiagnosticsDialog::DiagnosticsDialog(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::DiagnosticsDialog)
+{
+    ui->setupUi(this);
+
+    networkManager = new QNetworkAccessManager(this);
+    connect(networkManager, SIGNAL(finished(QNetworkReply *)), this, SLOT(getGithubVersionFinished(QNetworkReply *)));
+}
+
+DiagnosticsDialog::~DiagnosticsDialog()
+{
+    delete ui;
+}
+
+void DiagnosticsDialog::GetData() {
+    std::string cpiddata = GetListOf("beacon");
+    std::string sWhitelist = GetListOf("project");
+    int64_t superblock_age = GetAdjustedTime() - mvApplicationCacheTimestamp["superblock;magnitudes"];
+    double popularity = 0;
+    std::string consensus_hash = GetNeuralNetworkSupermajorityHash(popularity);
+    std::string sAge = ToString(superblock_age);
+    std::string sBlock = mvApplicationCache["superblock;block_number"];
+    std::string sTimestamp = TimestampToHRDate(mvApplicationCacheTimestamp["superblock;magnitudes"]);
+    printf("Pushing diagnostic data...");
+    double lastblockage = PreviousBlockAge();
+    double PORDiff = GetDifficulty(GetLastBlockIndex(pindexBest, true));
+    syncData = "<WHITELIST>" + sWhitelist + "</WHITELIST><CPIDDATA>"
+        + cpiddata + "</CPIDDATA><QUORUMDATA><AGE>" + sAge + "</AGE><HASH>" + consensus_hash + "</HASH><BLOCKNUMBER>" + sBlock + "</BLOCKNUMBER><TIMESTAMP>"
+        + sTimestamp + "</TIMESTAMP><PRIMARYCPID>" + KeyValue("cpid") + "</PRIMARYCPID><LASTBLOCKAGE>" + ToString(lastblockage) + "</LASTBLOCKAGE><DIFFICULTY>" + RoundToString(PORDiff,2) + "</DIFFICULTY></QUORUMDATA>";
+    testnet_flag = fTestNet ? "TESTNET" : "MAINNET";
+}
+
+int DiagnosticsDialog::VarifyBoincPath() {
+    boost::filesystem::path boincPath = (boost::filesystem::path) GetBoincDataDir();
+    if(boincPath.empty())
+        boincPath = KeyValue("boincdatadir");
+
+    if(boost::filesystem::is_directory(boincPath))
+        return 1;
+    else
+        return 0;
+}
+
+int DiagnosticsDialog::FindCPID() {
+    std::string cpid = KeyValue("cpid");
+
+    if(cpid.length() != 0)
+        return 1;
+    else
+        return 0;
+}
+
+int DiagnosticsDialog::VarifyIsCPIDValid() {
+    boost::filesystem::path clientStatePath = (boost::filesystem::path) GetBoincDataDir();
+    if(!clientStatePath.empty())
+        clientStatePath = clientStatePath / "client_state.xml";
+    else
+        clientStatePath = KeyValue("boincdatadir");
+
+    if(clientStatePath.empty())
+        return 0;
+
+    boost::filesystem::ifstream clientStateStream;
```

Comment:
CI has an issue here. u mean boost::filesystem::basic_ifstream ?

Reply:
Hmm I must be on a different boost version or something, are we still using 1.55?

Reply:
as far as i know yes however i have boost 1.62 on my system

Reply:
Ok might have to downgrade my local install then, think I've got 1.65, or whatever is latest in the arch repos. I did check everything compiled before making the pull request, honest :)